src/legacy/core_plugins/data/public/search/aggs/metrics/top_hit.ts

```diff
@@ -63,10 +63,7 @@ export const topHitMetricAgg = new MetricAggType({
       name: 'field',
       type: 'field',
       onlyAggregatable: false,
-      filterFieldTypes: (aggConfig: IMetricAggConfig) =>
-        _.get(aggConfig.schema, 'aggSettings.top_hits.allowStrings', false)
-          ? '*'
-          : KBN_FIELD_TYPES.NUMBER,
+      filterFieldTypes: '*',
```

Comment:
top hit supports all field types (aggregation). the fact that in most visualizations we only want number fields is visualization specific, handling that has been moved to default editor